timeseries/src/autogluon/timeseries/models/abstract/abstract_timeseries_model.py

```diff
@@ -171,6 +172,22 @@ def load_oof_predictions(cls, path: str, verbose: bool = True) -> List[TimeSerie
         """Load the cached OOF predictions from disk."""
         return load_pkl.load(path=os.path.join(path, "utils", cls._oof_filename), verbose=verbose)
 
+    @property
+    def supports_known_covariates(self) -> bool:
+        return self._create_covariate_regressor() is not None or self.__class__._supports_known_covariates
```

Comment:
I would recommend checking for something like `self._get_model_params().get("covariate_regressor") is not None` to avoid any potential side effects of  `_create_covariate_regressor()` (like #4832).

Reply:
I thought about this a bit. User-defined parameters are not a great "source of truth," as we have a habit of overriding them. It's not unrealistic that in the near future some models override user-provided "CAT" to None, or None to "CAT" or even override during deferred init. I agree regarding the side effect but I think this deserves some thought.

Reply:
I don't currently see such scenario (overriding user params for `covariate_regressor`) as very likely and have a slight preference for the hyperparameter check, but I'm also okay with the current solution - up to you.

Reply:
OK, will switch to hyperparameters for now.